### PR TITLE
[ios]  Add iOS bindings and example for missing icons event.

### PR DESCRIPTION
**Pull request by jmkiley**
Adds iOS bindings for #14253. 

cc @julianrex @ansis

### CODE REVIEW

File: platform/ios/src/MGLMapViewDelegate.h [hunk 1]
```diff
@@ -275,6 +275,8 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style;
 
+
+- (NSString *)mapViewMissingStyleImage:(NSString *)image;
```
Comment (fabian-guerra) [inline platform/ios/src/MGLMapViewDelegate.h]: Something along the lines of:
`-(NSString *)mapView:(MGLMapView *)mapView didFailLoadingImage:(NSString *)missingImage;`

Comment (julianrex) [reply platform/ios/src/MGLMapViewDelegate.h]: `...didFailToLoadImage:` ?

**Comment by jmkiley** [comment]
I started to implement this, but I was not able to finish [porting the example](https://github.com/mapbox/mapbox-gl-native/commit/ed923d5331e103dd46858fb998e2da76f6172776). I may have misunderstood what was supposed to happen, sorry!

**Comment by fabian-guerra** [comment]
This is how it's rendered on iOS with the substitute image:
cc @ansis @julianrex 
missing-icon

**Pull request merged**